backend/telemetry_core/src/main.rs

```diff
@@ -353,8 +353,7 @@ where
 {
     // unbounded channel so that slow feeds don't block aggregator progress:
     let (tx_to_feed_conn, rx_from_aggregator) = flume::unbounded();
-    let mut rx_from_aggregator_chunks =
-        ReadyChunksAll::new(common::flume_receiver_into_stream(rx_from_aggregator));
+    let mut rx_from_aggregator_chunks = ReadyChunksAll::new(rx_from_aggregator.into_stream());
```

Comment:
This is the remaining use of the potentially problematic `into_stream` method. In this case, we have a separate task which selects between items from this, and a close channel. Since the close channel only fires when we want to end, we hopefully don't `poll` the thing more than once until it's ready (suspending the task until it is).

Because the leak happens each time a stream is polled but not yet ready, my hope is that we'll be OK here.

I kept this use of the flume because we make use of metrics from these channels and I didn't want to have to remove those.

Reply:
(I reworded this into a comment in the code, since I think it's important to be aware of for now)